src/Model/ModelManagerInterface.php

```diff
@@ -35,35 +37,49 @@ public function getNewFieldDescriptionInstance($class, $name, array $options = [
     /**
      * @param object $object
      *
+     * @psalm-param T $object
+     *
      * @throws ModelManagerException
      */
     public function create($object);
 
     /**
      * @param object $object
      *
+     * @psalm-param T $object
+     *
      * @throws ModelManagerException
      */
     public function update($object);
 
     /**
      * @param object $object
      *
+     * @psalm-param T $object
+     *
      * @throws ModelManagerException
      */
     public function delete($object);
 
     /**
      * @param string $class
      *
-     * @return array all objects matching the criteria
+     * @return object[] all objects matching the criteria
+     *
+     * @psalm-template C
```

Comment:
Is `C` supposed to be a subclass of `T` here? If yes, shouldn't we have the following?
```suggestion
     * @psalm-template C of T
```

Reply:
I'm not sure if it must be a subclass or not